lib/oxrdfxml/src/error.rs

```diff
@@ -64,78 +31,45 @@ impl From<quick_xml::Error> for ParseError {
             quick_xml::Error::Io(error) => {
                 Self::Io(Arc::try_unwrap(error).unwrap_or_else(|e| io::Error::new(e.kind(), e)))
             }
-            _ => Self::Syntax(SyntaxError {
-                inner: SyntaxErrorKind::Xml(error),
-            }),
+            _ => Self::Syntax(SyntaxError::Xml(error)),
         }
     }
 }
 
 /// An error in the syntax of the parsed file.
-#[derive(Debug)]
-pub struct SyntaxError {
-    pub(crate) inner: SyntaxErrorKind,
-}
-
-#[derive(Debug)]
-pub enum SyntaxErrorKind {
-    Xml(quick_xml::Error),
+#[derive(Debug, thiserror::Error)]
+pub enum SyntaxError {
+    #[error(transparent)]
+    Xml(#[from] quick_xml::Error),
+    #[error("error while parsing IRI '{iri}': {error}")]
     InvalidIri {
         iri: String,
+        #[source]
         error: IriParseError,
     },
+    #[error("error while parsing language tag '{tag}': {error}")]
     InvalidLanguageTag {
         tag: String,
+        #[source]
         error: LanguageTagParseError,
     },
-    Msg {
-        msg: String,
-    },
+    #[error("{msg}")]
+    Msg { msg: String },
```

Comment:
The variant list is imho an implementation detail, I would prefer keeping it private (and I am not sure anyone would want to `match` on it)